File: 프로그래머스/lv2/42584. 주식가격/주식가격.cpp
#include <string>
#include <queue>
#include <vector>

using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer;
    int i, j, price_size = prices.size();
    
     for(i = 0; i < price_size - 1; i++){
        for(j = i + 1; j < price_size; j++){
            if(prices[i] > prices[j]){
                answer.push_back(j - i);
                break;
            }
        }
        if(answer.size() != i + 1){
            answer.push_back(j - i - 1);
        }
    }
    answer.push_back(0);
    
    return answer;
}

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        Stack<Integer> s = new Stack<>();
        
        int i=0;
        s.push(0);
        for(i=1; i<prices.length; i++){
            
            while(!s.empty() && prices[s.peek()] > prices[i]){
                answer[s.peek()] = i - s.peek();
                s.pop();
            }
            s.push(i);
        }
        while(!s.empty()){
            answer[s.peek()] = i - s.peek() - 1;
            s.pop();
        }
        
        return answer;
    }
}